tests/testthat/test-commit.R

```diff
@@ -72,3 +71,41 @@ test_that("creating a commit in another directory without author works", {
   sig <- git_signature_default(repo)
   expect_equal(log$author, git_signature_info(sig)$author)
 })
+
+test_that("status reports a conflicted file", {
+  # temporary measure until gert can do a non fast forward merge
+  skip_if_not_installed("git2r")
+
+  repo <- git_init(tempfile("gert-test-conflicts"))
+  on.exit(unlink(repo, recursive = TRUE))
+
+  foo_path <- file.path(repo, "foo.txt")
+
+  writeLines("cranky-crab-legs", foo_path)
+  git_add("foo.txt", repo = repo)
+  git_commit("Add a file", author = gert_testing_signature, repo = repo)
+
+  git_branch_create("my-branch", repo = repo)
+  writeLines("cranky-CRAB-LEGS", foo_path)
+  git_add("foo.txt", repo = repo)
+  git_commit("Uppercase last 2 words", author = gert_testing_signature, repo = repo)
+
+  git_branch_checkout("master", repo = repo)
+  writeLines("CRANKY-CRAB-legs", foo_path)
+  git_add("foo.txt", repo = repo)
+  git_commit("Uppercase first 2 words", author = gert_testing_signature, repo = repo)
+
+  # TODO: switch to a gert function when possible
+  # https://github.com/r-lib/gert/issues/41
+  when <- structure(list(time = 1395567947, offset = 60), class = "git_time")
+  git2r_sig <- structure(
+    list(name = "Testing Jerry", email = "test@jerry.com", when = when),
+    class = "git_signature"
+  )
```

Comment:
git2r doesn't offer a constructor for its `git_signature` object. It would probably be nicer to configure local user here?